Explorer/Assets/Plugins/DCLCrypto/Authenticator.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+
+namespace DCLCrypto
+{
+    public static class Authenticator
+    {
+        public delegate string SignerFunc(string payload);
+
+        public static string GetEphemeralMessage(string ephemeralAddress, DateTime expiration) =>
+            $"Decentraland Login\nEphemeral address: {ephemeralAddress}\nExpiration: {expiration:s}";
+
+        public static AuthIdentity CreateRandomInsecureAuthIdentity()
+        {
+            var identity = Identity.CreateRandom();
+            var ephemeralIdentity = Identity.CreateRandom();
+
+            SignerFunc signerFunc = payload => identity.Sign(payload);
+
+            return InitializeAuthChain(identity.Address(), signerFunc, ephemeralIdentity, 600);
+        }
+
+        public static AuthIdentity InitializeAuthChain(string ethAddress, SignerFunc signerFunc, Identity ephemeralIdentity, int ephemeralMinutesDuration)
+        {
+            DateTime expiration = DateTime.Now.AddMinutes(ephemeralMinutesDuration);
+
+            string ephemeralMessage = GetEphemeralMessage(ephemeralIdentity.Address(), expiration);
+            string firstSignature = signerFunc(ephemeralMessage);
+
+            var authChain = new AuthChain
+            {
+                new ()
+                {
+                    type = AuthLinkType.SIGNER,
+                    payload = ethAddress,
+                    signature = "",
+                },
+                new ()
+                {
+                    type = AuthLinkType.ECDSA_EPHEMERAL,
+                    payload = ephemeralMessage,
+                    signature = firstSignature,
+                },
+            };
+
+            return new AuthIdentity(ethAddress, authChain, ephemeralIdentity, expiration);
+        }
+
+        public static AuthChain SignPayload(AuthIdentity authIdentity, string entityId)
+        {
+            AuthChain authChain = authIdentity.AuthChain.Clone(); // new list, we must not modify the AuthIdentity
```

Comment:
`Clone` seems very impactful if called often. How often do we need to call it and what exactly are we going to modify from outside, considering that `AuthChain` is a mutable list?

I also worried that this is a temporary allocation just to get it, serialize and forget. And leave `GC` to handle the rest. We should improve it.

Reply:
We need to store the Initial AuthChain, which has 2 AuthLinks. The first element is the Identity as payload, and the other is for signing with the EphemeralKey. The third one is the payload that changes on each signed message.

The signature of each element must be signed with the key of the address of the previous signature. The first element doesn't require a signature; it must always be of type "AuthLinkType.SIGNER".

So, that's the reason that I'm copying the first two elements and adding one on the end.